routing/router.go

```diff
@@ -442,6 +486,35 @@ func (r *ChannelRouter) networkHandler() {
 			// after N blocks pass with no corresponding
 			// announcements.
 
+		case chainUpdate, ok := <-r.staleBlocks:
+			// If the channel has been closed, then this indicates
+			// the daemon is shutting down, so we exit ourselves.
+			if !ok {
+				return
+			}
+
+			// Since this block is stale, we update our best height
+			// to the previous block.
+			blockHeight := uint32(chainUpdate.Height)
+			r.bestHeight = blockHeight - 1
+
+			// Update the channel graph to reflect that this block
+			// was disconnected.
+			err := r.cfg.Graph.DisconnectBlockAtHeight(blockHeight)
+			if err != nil {
+				log.Errorf("unable to prune graph with stale "+
+					"block: %v", err)
+				continue
+			}
+
+			// Invalidate the route cache, as some channels might
+			// not be confirmed anymore.
+			r.routeCacheMtx.Lock()
+			r.routeCache = make(map[routeTuple][]*Route)
+			r.routeCacheMtx.Unlock()
+
+			// TODO(halseth): notify client about the reorg?
```

Comment:
Which client? Notification clients?

Reply:
Yeah, as for new blocks:
```go
// Notify all currently registered clients of the newly
// closed channels.
closeSummaries := createCloseSummaries(blockHeight, chansClosed...)
r.notifyTopologyChange(&TopologyChange{
        ClosedChannels: closeSummaries
})
```

Reply:
Ah yes, that should be done. This can be done isn a follow up PR/commit as this one is nearly ready to be merged.